qiskit_superstaq/custom_gates.py

```diff
@@ -183,6 +185,94 @@ def __str__(self) -> str:
         return f"ParallelGates({args})"
 
 
+class IITOFFOLIGate(ControlledGate):
+    def __init__(self, label: Optional[str] = None, ctrl_state: Optional[Union[str, int]] = None):
+        """Create new CCX gate."""
+        super().__init__(
+            "iitoffoli",
+            3,
+            [],
+            num_ctrl_qubits=2,
+            label=label,
+            ctrl_state=ctrl_state,
+            base_gate=XGate(),
+        )
+
+    def _define(self) -> None:
+        qc = qiskit.QuantumCircuit(3, name="iitoffoli")
+        csgate = SGate().control(1)
+        qc.x(0)
+        qc.x(1)
+        qc.ccx(0, 1, 2)
+        qc.append(csgate, [0, 1])
+        qc.x(0)
+        qc.x(1)
```

Comment:
this should take into account `ctrl_state`, e.g. something like
```suggestion
        if self.ctrl_state & 2:
            qc.x(0)
        if self.ctrl_state & 1:
            qc.x(1)
        qc.ccx(0, 1, 2)
        qc.append(csgate, [0, 1])
        if self.ctrl_state & 2:
            qc.x(0)
        if self.ctrl_state & 1:
            qc.x(1)
```

Reply:
Done.